planetarypraxis/smartforests
# Fix logbook thumbnails

Fixes #98
- Fixes the edge case of a logbook having three images in its` LogbookEntryPage`s and only rendering a thumbnail with 1 image in it.
- Considers ALL images in each `LogbookEntryPage` for thumbnail inclusion, rather than just the first one in each.
- Ensures that each unique image (i.e. database row) is only included once in the thumbnail
- Allows larger grids beyond 2x2 (e.g. 3x3 if you have 9 images in a logbook)
- Ensures that when a LogbookEntryPage is deleted, the parent page thumbnail is regenerated (via signal)

## How Can It Be Tested?
- Set up a logbook with 1 image in it, and see a thumbnail with only 1 image in it
- Add an additional 2 images inside it (for a total of 3), and see a thumbnail on the index page with only 2 images in it
  - Could be three separate logbookentries with 1 image each
  - Or One logbookentry with 3 images in
- Add an additional 1 image (for a total of 4) and see a thumbnail with a 2x2 grid in it
- (If you desire) add an additional 5 images and see a thumbnail with a 3x3 grid in it

## Types of changes


## Checklist:
